helmfile/values/thanos/objectstorage-secret.yaml.gotmpl

```diff
@@ -1,20 +1,49 @@
-{{ if not (eq .Values.objectStorage.type "s3") }}
-{{ fail "\nERROR: Thanos requires s3 object storage, see Values.objectStorage.type" }}
-{{ end }}
-
 query:
   enabled: false
 
 queryFrontend:
   enabled: false
 
 objstoreConfig: |-
-  {{- if eq .Values.objectStorage.type "s3" }}
+  {{- $type := default .Values.objectStorage.type .Values.thanos.objectStorage.type }}
+
+  {{- if and (ne $type "s3") (ne $type "swift") }}
+  {{- fail "\nERROR: Thanos requires either S3 or Swift object storage, see Values.objectStorage.type" }}
+  {{- end }}
+
+  {{- if eq $type "s3" }}
   type: s3
   config:
     bucket: {{ .Values.objectStorage.buckets.thanos }}
     endpoint: {{ trimPrefix "https://" (trimPrefix "http://" .Values.objectStorage.s3.regionEndpoint) }}
     access_key: {{ .Values.objectStorage.s3.accessKey }}
     secret_key: {{ .Values.objectStorage.s3.secretKey }}
     region: {{ .Values.objectStorage.s3.region }}
+
+  {{- else if eq $type "swift" }}
+  {{- $swift := .Values.objectStorage | getOrNil "swift" | required "Swift enabled for Thanos but not configured!" }}
+  type: swift
+  config:
+    auth_version: {{ $swift.authVersion }}
+    auth_url: {{ $swift.authUrl }}
+    {{- if hasKey $swift "region" }}
```

Comment:
Do we want to add these extra checks that we have here to the harbor config as well?

Reply:
Some are a requirement for harbor but not for thanos/objstore.

Reply:
Right :+1: